internal/services/automation/automation_runtime_environment_resource.go

```diff
@@ -0,0 +1,306 @@
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return meta.ResourceRequiresImport(m.ResourceType(), id)
+			}
+
+			req := runtimeenvironment.RuntimeEnvironment{}
+			req.Properties = &runtimeenvironment.RuntimeEnvironmentProperties{
+				Runtime: &runtimeenvironment.RuntimeProperties{
+					Language: pointer.FromString(model.RuntimeLanguage),
+					Version:  pointer.FromString(model.RuntimeVersion),
+				},
+			}
+			req.Location = azure.NormalizeLocation(model.Location)
+
+			if model.Tags != nil {
+				req.Tags = &model.Tags
+			}
+
+			if model.RuntimeDefaultPackages != nil {
+				req.Properties.DefaultPackages = pointer.To(model.RuntimeDefaultPackages)
+			}
+
+			if model.Description != "" {
+				req.Properties.Description = pointer.FromString(model.Description)
+			}
+
+			if _, err = client.Create(ctx, id, req); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			meta.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (m AutomationRuntimeEnvironmentResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.Automation.RuntimeEnvironment
+
+			id, err := runtimeenvironment.ParseRuntimeEnvironmentID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return meta.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := AutomationRuntimeEnvironmentResourceModel{
+				Name: id.RuntimeEnvironmentName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.ResourceGroupName = id.ResourceGroupName
+				state.AutomationAccountName = id.AutomationAccountName
+				state.RuntimeLanguage = pointer.ToString(resp.Model.Properties.Runtime.Language)
+				state.RuntimeVersion = pointer.ToString(resp.Model.Properties.Runtime.Version)
+				state.Location = resp.Model.Location
+
+				if resp.Model.Properties.DefaultPackages != nil {
+					state.RuntimeDefaultPackages = *resp.Model.Properties.DefaultPackages
+				}
+
+				if resp.Model.Properties.Description != nil {
+					state.Description = pointer.ToString(resp.Model.Properties.Description)
+				}
+
+				if resp.Model.Tags != nil {
+					state.Tags = *resp.Model.Tags
+				}
+			}
+
+			return meta.Encode(&state)
+		},
+	}
+}
+
+func (m AutomationRuntimeEnvironmentResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) (err error) {
+			client := meta.Client.Automation.RuntimeEnvironment
+
+			id, err := runtimeenvironment.ParseRuntimeEnvironmentID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state AutomationRuntimeEnvironmentResourceModel
+			if err = meta.Decode(&state); err != nil {
+				return err
```

Comment:
nit: for consistency with the rest of the provider can we wrap this error

```suggestion
				return fmt.Errorf("decoding: %+v", err)
```